iris-mpc-cpu/src/hnsw/searcher.rs

```diff
@@ -181,6 +200,49 @@ impl Default for HnswSearcher {
 
 #[allow(non_snake_case)]
 impl HnswSearcher {
+    /// Two-step variant of `connect_bidir`: prepare.
+    async fn connect_prepare<V: VectorStore, G: GraphStore<V>>(
+        &self,
+        vector_store: &mut V,
+        graph_store: &G,
+        q: &V::VectorRef,
+        mut neighbors: FurthestQueueV<V>,
+        lc: usize,
+    ) -> ConnectPlanLayerV<V> {
+        let M = self.params.get_M(lc);
+        let max_links = self.params.get_M_max(lc);
+
+        neighbors.trim_to_k_nearest(M);
+
+        // Connect all n -> q.
+        let mut n_links = vec![];
+        for (n, nq) in neighbors.iter() {
+            let mut links = graph_store.get_links(n, lc).await;
+            links.insert(vector_store, q.clone(), nq.clone()).await;
+            links.trim_to_k_nearest(max_links);
+            n_links.push(links);
+        }
+
+        ConnectPlanLayer { neighbors, n_links }
+    }
+
+    /// Two-step variant of `connect_bidir`: execute.
+    async fn connect_apply<V: VectorStore, G: GraphStore<V>>(
```

Comment:
nit: Would it make sense to leave a to-do to refactor `connect_bidir` using these new functions?

Reply:
Done!